samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java

```diff
@@ -182,7 +182,13 @@ public CompletableFuture<SnapshotIndex> getSnapshotIndex(String blobId, Metadata
     return FutureUtil.executeAsyncWithRetries(opName, () -> {
       ByteArrayOutputStream indexBlobStream = new ByteArrayOutputStream(); // no need to close ByteArrayOutputStream
       return blobStoreManager.get(blobId, indexBlobStream, metadata, getDeleted).toCompletableFuture()
-          .thenApplyAsync(f -> snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray()), executor);
+          .thenApplyAsync(f -> {
+            try {
+              return snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray());
+            } catch (Exception e) {
+              throw new SamzaException(String.format("Unable to deserialize SnapshotIndex bytes for blob ID: %s", blobId), e);
```

Comment:
could we use `.handle` for the exception instead?

Reply:
Done. thanks